core/src/main/java/org/bitcoinj/core/EncodedPrivateKey.java

```diff
@@ -26,31 +28,46 @@
  * Some form of string-encoded private key. This form is useful for noting them down, e.g. on paper wallets.
  */
 public abstract class EncodedPrivateKey {
-    protected final NetworkParameters params;
+    protected final Network network;
     protected final byte[] bytes;
 
-    protected EncodedPrivateKey(NetworkParameters params, byte[] bytes) {
-        this.params = checkNotNull(params);
+    protected EncodedPrivateKey(Network network, byte[] bytes) {
+        this.network = checkNotNull(network);
         this.bytes = checkNotNull(bytes);
     }
 
+    @Deprecated
+    protected EncodedPrivateKey(NetworkParameters params, byte[] bytes) {
+        this(checkNotNull(params).network(), checkNotNull(bytes));
+    }
+
+    /**
+     * Get the network this data is prefixed with.
+     * @return the Network.
+     */
+    public Network network() {
+        return network;
+    }
+
     /**
      * @return network this data is valid for
+     * @deprecated Use {@link #network()}
      */
+    @Deprecated
     public final NetworkParameters getParameters() {
-        return params;
+        return NetworkParameters.of(network);
```

Comment:
This adds an indirect dependency on `Networks.get()`

Reply:
Yes, but it's deprecated so it will go away in the next release.